engine/crates/engine/src/lib.rs

```diff
@@ -1,44 +1,6 @@
 //! # A GraphQL server library implemented in Rust
-#![deny(clippy::all)]
-#![deny(clippy::inefficient_to_string)]
-#![deny(clippy::match_wildcard_for_single_variants)]
-#![deny(clippy::redundant_closure_for_method_calls)]
-#![deny(unused_crate_dependencies)]
-#![allow(clippy::cast_lossless)]
-#![allow(clippy::cast_possible_truncation)]
-#![allow(clippy::cast_sign_loss)]
-#![allow(clippy::cognitive_complexity)]
-#![allow(clippy::default_trait_access)]
-#![allow(clippy::doc_markdown)]
-#![allow(clippy::explicit_iter_loop)]
-#![allow(clippy::future_not_send)]
-#![allow(clippy::if_not_else)]
-#![allow(clippy::iter_without_into_iter)]
-#![allow(clippy::implicit_hasher)]
-#![allow(clippy::map_flatten)]
-#![allow(clippy::map_unwrap_or)]
-#![allow(clippy::manual_let_else)]
-#![allow(clippy::match_same_arms)]
-#![allow(clippy::missing_const_for_fn)]
-#![allow(clippy::missing_errors_doc)]
-#![allow(clippy::module_name_repetitions)]
-#![allow(clippy::must_use_candidate)]
-#![allow(clippy::needless_borrow)]
-#![allow(clippy::needless_pass_by_value)]
-#![allow(clippy::needless_lifetimes)]
-#![allow(clippy::option_if_let_else)]
 #![allow(clippy::panic)]
```

Comment:
why leave this?

Reply:
The ones that are left are those that need to be different in the engine crate (compared to the workspace rules). I could remove these three `allow`s but that would involve changing enum names, adding a bunch of lifetimes, and seeing how to replace the panics. I'm happy to do it if that sounds good. Especially the elided lifetimes I don't like, but it's probably a few hundred more lines of diff (`cargo clippy --fix` can't do it automatically,  I checked).